### ISSUE TITLE
decode URI component

### ISSUE THREAD

**Issue by AvengingSyndrome** [tfausak/blunt#25]
When loading from the window.location.hash decode location as a URI component. This allows users the ability to add it as a search engine to chrome. 

Currently doing a search like
"Pointfree: \x y -> x + y " becomes "%5Cx%20y%20-%3E%20x%20+%20y"

I think the change is window.location.hash.substring(7) -> decodeURIComponent(window.location.hash.substring(7));

**Comment by tfausak** [comment]
This is a weird one. I did most of my testing with Firefox 37. Both permalinks and keyword searches work with it. I just tried in Chrome and I was able to reproduce this bug after adding Blunt as a search engine. (Permalinks work fine as is in Chrome.) 

I don't think I can blindly call `decodeURIComponent`. For example, [`f x = x%20`](http://blunt.herokuapp.com/#input=f x = x%20) is a valid input that doesn't get stored correctly in the URI. I need to encode the input before putting it in the URL.
